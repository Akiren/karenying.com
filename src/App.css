@import url('https://fonts.googleapis.com/css?family=Varela+Round');

.App {
  text-align: center;
  background-color: #333333;
  padding: 0px;
  margin: 0px;
  min-height: 100vh;
  width: 100%;
}

* {
  font-family: 'Varela Round', sans-serif !important;
}