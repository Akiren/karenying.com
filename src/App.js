import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import './App.css';
import Home from './Components/Home';
import About from './Components/About';
import Projects from './Components/Projects';
import Contact from './Components/Contact';
import Error from './Components/Error';
import FooterIcon from './Components/FooterIcon';
import Footer from './Components/Footer';

class App extends React.Component {
    state = {
        footerActive: false,
    };

    handleFooterIconClick = () => {
        this.setState({ footerActive: true });
    };

    handleFooterCloseClick = () => {
        this.setState({ footerActive: false });
    };

    render() {
        return (
            <Router>
                <div className='App'>
                    <Switch>
                        <Route exact={true} path='/home' component={Home} />
                        <Route exact={true} path='/' component={Home} />
                        <Route exact={true} path='/about' component={About} />
                        <Route
                            exact={true}
                            path='/projects'
                            component={Projects}
                        />
                        <Route
                            exact={true}
                            path='/contact'
                            component={Contact}
                        />
                        <Route path='*' component={Error} />
                    </Switch>
                    {this.state.footerActive ? (
                        <Footer clickFunction={this.handleFooterCloseClick} />
                    ) : (
                        <FooterIcon
                            clickFunction={this.handleFooterIconClick}
                        />
                    )}
                </div>
            </Router>
        );
    }
}

export default App;
