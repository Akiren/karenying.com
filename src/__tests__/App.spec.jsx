import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

describe('App', () => {
  it('should render 2 react router Switches', () => {
    const wrapper = shallow(<App />);

    expect(wrapper.find('Switch').length).toEqual(1);
  });
});
