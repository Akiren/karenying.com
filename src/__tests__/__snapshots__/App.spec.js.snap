// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      path="/:page"
      render={[Function]}
    />
    <Redirect
      to="/about"
    />
  </Switch>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/about"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/projects"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/contact"
    />
    <Route
      component={[Function]}
      path="*"
    />
  </Switch>
  <Footer />
</BrowserRouter>
`;
